plugins/inventory/now.py

```diff
@@ -0,0 +1,443 @@
+                            type: list
+                            default: None
+"""
+
+EXAMPLES = r"""
+# A trivial example that creates a host from every record of the
+# ServiceNow cmdb_ci_server table. The ip_address column is used for
+# for ansible host, and server name for inventory hostname.
+# No groups will be created - all the resulting hosts are ungrouped.
+plugin: servicenow.itsm.now
+
+# `ansible-inventory -i inventory.now.yaml --graph` output:
+# @all:
+#  |--@ungrouped:
+#  |  |--DatabaseServer1
+#  |  |--DatabaseServer2
+#  |  |--INSIGHT-NY-03
+#  |  |--MailServerUS
+#  |  |--VMWARE-SD-04
+
+
+# Group hosts automatically, according to values of manufacturer and os columns.
+# Include only records with the specified operating systems.
+# Groups will most likely overlap.
+plugin: servicenow.itsm.now
+group_by:
+  manufacturer:
+  os:
+    includes:
+      - Linux Red Hat
+      - Windows XP
+
+# `ansible-inventory -i inventory.now.yaml --graph` output:
+# @all:
+#  |--@Dell_Inc_:
+#  |  |--DatabaseServer1
+#  |  |--DatabaseServer2
+#  |  |--INSIGHT-NY-03
+#  |--@Lenovo:
+#  |  |--FileServerFloor1
+#  |  |--FileServerFloor2
+#  |--@Linux_Red_Hat:
+#  |  |--DatabaseServer1
+#  |  |--DatabaseServer2
+#  |--@Windows_XP:
+#  |  |--FileServerFloor1
+#  |  |--FileServerFloor2
+#  |  |--INSIGHT-NY-03
+#  |--@ungrouped:
+
+
+# Group hosts into named groups, according to the specified criteria.
+# Below example creates a single group containing hosts that match
+# all the criteria.
+groups:
+  non_windows_prod_servers:
+    classification:
+      includes: [ Production ]
+    os:
+      excludes:
+        - Windows XP
+        - Windows 2000
+        - Windows 2000 Server
+        - Windows 2003 Standard
+
+# `ansible-inventory -i inventory.now.yaml --graph` output:
+# @all:
+#  |--@HP_UX:
+#  |  |--Car-2
+#  |--@Linux_Red_Hat:
+#  |  |--DatabaseServer1
+#  |  |--DatabaseServer2
+#  |  |--lnux100
+#  |  |--lnux101
+#  |--@Solaris:
+#  |  |--MailServerUS
+#  |--@ungrouped:
+
+
+# Configure inventory host names and host vars.
+plugin: servicenow.itsm.now
+columns:
+  - name
+  - classification
+  - cpu_type
+ansible_host_source: fqdn
+inventory_hostname_source: asset_tag
+
+# `ansible-inventory -i inventory.now.yaml --graph --vars` output:
+# @all:
+#  |--@ungrouped:
+#  |  |--P1000019
+#  |  |  |--{ansible_host = my.server.com}
+#  |  |  |--{classification = Production}
+#  |  |  |--{cpu_type = Intel}
+#  |  |  |--{name = SAP-SD-02}
+"""
+
+
+class InventoryModule(BaseInventoryPlugin):
+
+    NAME = "servicenow.itsm.now"
+
+    def verify_file(self, path):
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(("now.yaml", "now.yml")):
+                return True
+            # TODO remove?
+            self.display.vvv(
+                'Skipping due to inventory source not ending in "now.yaml" nor "now.yml"'
+            )
+            return False
```

Comment:
This should be dedented one level. Right now, if parent's class verification returns `False`, we exit with `None`, which is not the worst thing in the world, but also not the best ;)